### ISSUE TITLE
Compile fails in freetype due to HAVE_UNISTD_H not being defined

### ISSUE THREAD

**Issue by Imagix** [copperspice/copperspice#215]
Compiling copperspice on Mac OS 12.0.1 and Xcode 13.1 fails with:
```
/Users/andre/Development/copperspice/src/3rdparty/freetype/src/builds/unix/ftsystem.c:344:22: error: implicit declaration of function 'read' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        read_count = read( file,
                     ^
```
This seems to be able to be tracked back to HAVE_UNISTD_H not being defined and thus #include <unistd.h> isn't happening, which results in read, fread, and close all being undeclared identifiers.

**Comment by bgeller** [comment]
Thanks for reporting, we will investigate.

**Comment by Imagix** [comment]
Was diagnosing this a little more.  I pulled down a standalone copy of freetype and ran cmake on that to see what may be different between the two.   Turns out that the generated freetype/include/freetype/config/ftconfig.h is different in that in the standalone copy, it is where HAVE_UNISTD_H (and HAVE_FCNTL_H) are being #defined.  The copy of ftconfig.h in copperspice appears to be a static one which does not have those #defines.  I am unsure if it is safe to add those into the static copy (as that may impact every other platform that copperspice compiles on), or whether there is some standard mechanism that copperspice may be using to propogate its own header file discovery results into the 3rd party libs.

**Comment by agserm** [comment]
Can you please try the latest commit on the `m1-test` branch and see if the changes in freetype resolve this issue?

**Comment by Imagix** [comment]
Turns out, I got a new M1 Max MacBook after Cppcon, so I tried the m1-test branch.  Seems that this now compiles (as well as the version file issue) on the new M1.    Xcode 13.2.1, clang 13.0.0

**Comment by agserm** [comment]
Thanks so much for testing this, glad you got some new hardware to test the ARM side of things as well. Leaving this ticket open until the changes are on the master branch.

**Comment by agserm** [comment]
Fixed in d1f5261, thanks for your report.
